### ISSUE TITLE
Unity 2018.4.10f1 / Android Configuration Setup Error

### ISSUE THREAD

**Issue by JungPruna** [googlesamples/unity-jar-resolver#284]
Hello.
I'm trying to use Firebase Authentication with GPGS, but cannot make them works together.

Tested with Unity 2018.4.10f1 new project.
Trying to "Google Play Games - Android Configuration" setup.

1.2.105 Fine (But error with recent firebase sdk when resolve)
1.2.123 Fine (But error with recent firebase sdk when resolve)
1.2.124 Fine (But error with recent firebase sdk when resolve) 
1.2.125 Invalid classname: Method VersionHandlerImpl.UpdateVersionedAssets not found
1.2.129 Invalid classname: Method VersionHandlerImpl.UpdateVersionedAssets not found

And sometimes "Ambiguous match found" appears. (Not sure how to reproduce)

**Comment by ikitae** [comment]
me too

**Comment by sonhoang2610** [comment]
This happened to me too

**Comment by vnquanghvktqs** [comment]
me too, help me please

**Comment by reddtoric** [comment]
During the installation/import of the plugins (GPGS, updated play service resolver, or firebase) that had the VersionHandlerImpl error, did the import window in Unity show any files (e.g. files in PlayServiceResolver) that are obsolete and will be replaced? If so, you have to cancel the import, delete the old files and then do your import. Somewhere in the docs in GPGS or the resolver, it says you have to delete old files before importing new ones.

**Comment by vnquanghvktqs** [comment]
@reddtoric  : I used the latest PlayServiceResolver. Deleted all Google Play Games and re-enter it is still not possible. :(

**Comment by vnquanghvktqs** [comment]
Resolved on play-services-resolver-1.2.124.0 release!.

**Comment by firegate666** [comment]
I have the same issue in 1.2.129.0

**Comment by andoprof** [comment]
Unity 2018.4.11f1
Firebase 6.6.0
Google Play Game Services 0.9.4, 0.10.4
Have same issue!
Please fix, thanks!

**Comment by nishatoma** [comment]
Same issue here:
Unity 2019.2.7f2
Invalid classname: Method
VersionHandlerImpl.UpdateVersionedAssets not found

**Comment by stewartmiles** [comment]
If you're seeing invalid classname `Method VersionHandlerImpl.UpdateVersionedAssets` not found, you have an old version of the plugin in your project somewhere.  If you've been moving files around in your plugin then you should delete all instances of Play Services Resolver components from your project and reimport.  If you want to test that `VersionHandlerImpl.UpdateVersionedAssets` is present and working, I recommend creating an empty project, importing the latest release and waiting for it to enable all components since that process will also call `VersionHandlerImpl.UpdateVersionedAssets`.

**Comment by andoprof** [comment]
I've just tried Blank Project
Unity 2018.4.11f1
Firebase 6.6.0
Google Play Game Services 0.9.4, 0.10.4
Have same issue in 1.2.129:
Invalid classname: Method
VersionHandlerImpl.UpdateVersionedAssets not found
Please help, many thanks!

**Comment by stewartmiles** [comment]
@andoprof what order did you import the plugins?  If you overwrite the latest version of Google.VersionHandler.dll you may run into this sort of error.

**Comment by andoprof** [comment]
@stewartmiles 
Last time i've tried import the plugins in blank projects Unity 2018.4.11.f1
1. Play Services Resolver 1.2.129
2. Google Play Game Services 0.10.4
Have same issue in log:

Exception: Method VersionHandlerImpl.UpdateVersionedAssets not found
Google.VersionHandler.InvokeMethod (System.Type type, System.Object objectInstance, System.String methodName, System.Object[] args, System.Collections.Generic.Dictionary2[TKey,TValue] namedArgs) (at /Users/smiles/dev/src/unity-jar-resolver/source/VersionHandler/src/VersionHandler.cs:554) Google.VersionHandler.InvokeStaticMethod (System.Type type, System.String methodName, System.Object[] args, System.Collections.Generic.Dictionary2[TKey,TValue] namedArgs) (at /Users/smiles/dev/src/unity-jar-resolver/source/VersionHandler/src/VersionHandler.cs:485)
Google.VersionHandler.InvokeImplMethod (System.String methodName, System.Object[] args, System.Collections.Generic.Dictionary`2[TKey,TValue] namedArgs, System.Boolean schedule) (at /Users/smiles/dev/src/unity-jar-resolver/source/VersionHandler/src/VersionHandler.cs:416)
Google.VersionHandler.UpdateVersionedAssets (System.Boolean forceUpdate) (at /Users/smiles/dev/src/unity-jar-resolver/source/VersionHandler/src/VersionHandler.cs:383)
GooglePlayGames.Editor.GPGSAndroidSetupUI.PerformSetup (System.String clientId, System.String classDirectory, System.String className, System.String resourceXmlData, System.String nearbySvcId) (at Assets/GooglePlayGames/Editor/GPGSAndroidSetupUI.cs:121)
UnityEngine.Debug:LogException(Exception)
GooglePlayGames.Editor.GPGSAndroidSetupUI:PerformSetup(String, String, String, String, String) (at Assets/GooglePlayGames/Editor/GPGSAndroidSetupUI.cs:125)
GooglePlayGames.Editor.GPGSAndroidSetupUI:DoSetup() (at Assets/GooglePlayGames/Editor/GPGSAndroidSetupUI.cs:330)
GooglePlayGames.Editor.GPGSAndroidSetupUI:OnGUI() (at Assets/GooglePlayGames/Editor/GPGSAndroidSetupUI.cs:302)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)

**Comment by reddtoric** [comment]
@andoprof 
Try importing GPGS 0.10.4 first then Play Service Resolver 1.2.129. 
GPGS 0.10.4 uses Play Service Resolver 1.2.124 and in Stewartmiles' words, 

> If you overwrite the latest version of Google.VersionHandler.dll you may run into this sort of error.

<br>

@stewartmiles (Correct me if I'm wrong)  
Everyone experiencing such issues should try this:  
1. Figure out what version of Play Service Resolver each plugin uses
2. Import the plugins with Play Service Resolver from most outdated to latest version

**Comment by nishatoma** [comment]
To get around this issue for now, using version 0.9.64 of the play games plugin works.

**Comment by DavidZobrist** [comment]
![Error](https://user-images.githubusercontent.com/44160861/67272652-3cf20100-f4bd-11e9-98cb-a6bfc1810806.PNG)




Same issue. Still got active resource with achievement information in there.
And if I build the apk via cloud build the google play services seem to connect.

But i can not modifiy 
Window -> Google Play Games-> Setup -> Android Setup

If i hit the setup button I get:
VersionHandlerImpl.UpdateVersionedAssets not found!



tried:
Update to the newest stable build of Unity 2019.2.9f1
Update Google play services to v 0.10.04
Updated Firebase crashlytics and analytics v 6.6.0
Reinstalled the Jar resolver v. 1.2.129

**Comment by konidk** [comment]
Same here, happens with:

1. Play Services Resolver 1.2.125
2. Added GPGS 0.10.04 and Play Services Resolver 1.2.129 contained in the same unitypackage. I cleaned all previous references to Play Services Resolver 1.2.125 afterwards
3. **Issue happened**
4. Added GPGS 0.9.64 as stated by @nishatoma 
3. **Issue happened again**

Seems to be related to Play Services Resolver as @reddtoric says

**Comment by 0Cristofer** [comment]
I tested some things and posted in a Play Services issue thread, here it is:
[(https://github.com/playgameservices/play-games-plugin-for-unity/issues/2753#issuecomment-545258860)]

**Comment by stewartmiles** [comment]
Folks, to get on the same page here it would be useful to share a minimal project that demonstrates this behavior.  As I mentioned before, if Google.VersionHandler.dll is old this will likely happen so it would be neat to compare the checksum of the file in a problematic project vs. a working project.

**Comment by PixelEnvision** [comment]
This is the same issue I've posted earlier https://github.com/googlesamples/unity-jar-resolver/issues/283

I've found the issue is caused by Google.VersionHandlerImpl_v1.2.129.0 file in resolver.

I've tried the following and it works:
1. Import resolver 1.2.129
2. if you call "Google.VersionHandler.UpdateVersionedAssets(forceUpdate: true);" it will fail.
3. Import resolver 1.2.124 and only select 4 Google.VersionHandler files.
4. When asked select cancel to delete old versions
5. Delete Google.VersionHandlerImpl_v1.2.129.0 files
6. Rename Google.VersionHandlerImpl_v1.2.124.0 files to Google.VersionHandlerImpl_v1.2.129.0 (not sure if thats needed as I didn't try otherwise)
7. if you call "Google.VersionHandler.UpdateVersionedAssets(forceUpdate: true);" it will work now.

Only change in project was the above, no change to GPGS, etc... I hope this helps...
